go/vt/automation/resharding_task.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package automation
+
+import (
+	"fmt"
+	"strings"
+
+	pb "github.com/youtube/vitess/go/vt/proto/automation"
+	"github.com/youtube/vitess/go/vt/topo"
+)
+
+// ReshardingTask is a cluster operation which allows to increase the number of shards.
```

Comment:
ACK

Ammar suggested to rename it to HorizontalReshardingTask. When we support merges, there will be another task.